src/printer.js

```diff
@@ -133,31 +180,74 @@ function printMemberChain(path, options, print) {
   //    PropertyLookup d, Call]
   const printedNodes = [];
 
-  function traverse(path) {
+  // Here we try to retain one typed empty line after each call expression or
```

Comment:
im not really sure what this function is doing from this comment - can you explain a little more? maybe an example would be helpful